README.md

```diff
@@ -1,29 +1,131 @@
+  Counters.Counter private _tokenIds;
 
-...
+  constructor() ERC721("Bored Pets Yacht Club", "BPYC") {}
+
+  function mint(string memory _tokenURI) public {
+    _tokenIds.increment();
+    uint256 newTokenId = _tokenIds.current();
+    _safeMint(msg.sender, newTokenId);
+    _setTokenURI(newTokenId, _tokenURI);
+  }
+}
+```
+Let's take a look at the imports:
+
+- `@openzeppelin/contracts/token/ERC721/ERC721.sol`
+
+To be a valid NFT, `BoredPetsNFT` implements the ERC-721 standard by inheriting the implementation of `ERC721URIStorage.sol`
+
+- `@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol`
+
+`ERC721URIStorage.sol` is an extension of `ERC721.sol`, which stores the `tokenURI` in `storage` (aka on-chain). If we did not use this extension, the `tokenURI` is derived by concatenating a `baseURI` and `tokenId`. However, we want to store the metadata on IPFS, which identifies content via a hash. Therefore, we cannot generate the `tokenURI` by using the `tokenId`. Instead, we must store the `tokenURI` on-chain to get that data.
+
+To use this, we simply need to code `BoredPetsNFT is ERC721URISTorage`.
+
+- `@openzeppelin/contracts/utils/Counters.sol`
 
-## Front End
+In order to guarantee each `tokenId` is unique, we will increment it by one each time a NFT is minted. We use `Counters.Counter private _tokenIds` to increment and track the value of `tokenId`.
+
+Now, let's go over the methods:
+
+- `constructor() ERC721("Bored Pets Yacht Club", "BPYC")`
+
+The constructor is called when we deploy the smart contract. In this case, `constructor()` indicates our our NFT contract does not take in any parameters. Then, we set "Bored Pets Yacht Club" as the name and "BPYC" as the symbol for our NFTs.
+
+- `function mint(string memory _tokenURI) public`
+
+This funtion is super straightforward! Every time we mint an NFT, we want to guarantee `tokenId` is unique by incrementing the counter by one. Then, we just call `_safeMint` to mint and store the `_tokenURI`!
+
+## Write the deployment script
+
+In order to deploy the smart contract, we'll need to write our migrations file. Create the file `1_deploy_contracts.js` under `packages/truffle/migrations`. Then, copy and paste the code below:
+
+```javascript
+const BoredPetsNFT = artifacts.require("BoredPetsNFT");
+
+module.exports = function (deployer) {
+  deployer.deploy(BoredPetsNFT);
+};
+```
+`artifacts.require` is how we tell Truffle which contracts we'd like to interact with and returns a contract abstraction that we can use in the rest of our deployment script. NOTE: The name should match the contract name (`BoredPetsNFT`) and NOT the filename (`BoredPetsNFT.sol`).
+
+## Deploy the NFT smart contract
+
+This tutorial outlines how to deploy with Infura, but Truffle has great features to speed up local testing:
+
+1. Use Dashboards so you don't have to expose your private key
+2. Use Ganache forking to test using real data and interact with live contracts locally
```

Comment:
As someone who is not an expert on Ganache, the following sentence reads funny to me.

"Use Ganache forking to test using real data and interact with live contracts locally"

Is there a way to be more clear even if it takes a few extra words. Or maybe we link to what Ganache forking is (maybe you have done that somewhere else and I have missed it?) Also is "live contracts locally" the best way to describe this? Maybe it's just the word "live" throwing me off. 

Does the word "live" need to be there?